File: BE_QuanLyTro/src/main/java/ut/edu/be_quanlytro/Dto/Request/UserUpdateRequest.java
package ut.edu.be_quanlytro.Dto.Request;

import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Getter @Setter
public class UserUpdateRequest {
    private String phone;
    private String password;
    private String fullName;
    private LocalDate dob;
    private String hometown;
}

File: Flutter_QuanLyTro/lib/data/models/request/user_update_request.dart
class UserUpdateRequest {
  final String phone;
  final String? password;
  final String fullName;
  final String dob;
  final String hometown;

  UserUpdateRequest({
    required this.phone,
    this.password,
    required this.fullName,
    required this.dob,
    required this.hometown,
  });

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = {
      'phone': phone,
      'fullName': fullName,
      'dob': dob,
      'hometown': hometown,
    };

    if (password != null && password!.isNotEmpty) {
      data['password'] = password;
    }

    return data;
  }
}